src/features/runtime-checks.js

```diff
@@ -0,0 +1,249 @@
+            proto = Object.getPrototypeOf(proto)
+        }
+        propertyNames.forEach(prop => {
+            if (prop === 'constructor') return
+            Object.defineProperty(this, prop, {
+                get () {
+                    return el[prop]
+                },
+                set (value) {
+                    el[prop] = value
+                }
+            })
+        })
+    }
+
+    /**
+     * The element has been moved to the DOM, so we can now reflect all changes to a real element.
+     * This is to allow us to interrogate the real element before it is moved to the DOM.
+     */
+    transplantElement () {
+        // Creeate the real element
+        const el = initialCreateElement.call(document, this.#tagName)
+
+        // Reflect all attrs to the new element
+        for (const attribute of this.getAttributeNames()) {
+            el.setAttribute(attribute, this.getAttribute(attribute))
+        }
+
+        // Reflect all props to the new element
+        for (const param of Object.keys(this)) {
+            el[param] = this[param]
+        }
+
+        // Reflect all listeners to the new element
+        for (const [...args] of this.#listeners) {
+            el.addEventListener(...args)
+        }
+        this.#listeners = []
+
+        // Move all children to the new element
+        while (this.firstChild) {
+            el.appendChild(this.firstChild)
+        }
+
+        // Move the new element to the DOM
+        try {
+            this.insertAdjacentElement('afterend', el)
+        } catch (e) { console.warn(e) }
+
+        this.monitorProperties(el)
+        // TODO pollyfill WeakRef
+        this.#el = new WeakRef(el)
+
+        // Delay removal of the custom element so if the script calls removeChild it will still be in the DOM and not throw.
+        setTimeout(() => {
+            this.remove()
+        }, elementRemovalTimeout)
+    }
+
+    getElement () {
+        return this.#el?.deref()
+    }
+
+    setAttribute (name, value) {
+        const el = this.getElement()
+        if (el) {
+            return el.setAttribute(name, value)
+        }
+        return super.setAttribute(name, value)
+    }
+
+    removeAttribute (name) {
+        const el = this.getElement()
+        if (el) {
+            return el.removeAttribute(name)
+        }
+        return super.removeAttribute(name)
+    }
+
+    addEventListener (...args) {
+        const el = this.getElement()
+        if (el) {
+            return el.addEventListener(...args)
+        }
+        this.#listeners.push([...args])
+    }
+
+    removeEventListener (...args) {
+        const el = this.getElement()
+        if (el) {
+            return el.removeEventListener(...args)
+        }
+        this.#listeners = this.#listeners.filter((listener) => {
+            return listener[0] !== args[0] || listener[1] !== args[1]
+        })
+    }
+
+    toString () {
+        // TODO
```

Comment:
Is there more to do here? If so, maybe add a description of what this should do.

Reply:
Fixed by translating the first letter to upper case, there will be some edge cases that might be wrong but that's fine for now as we're only looking to support script for now.